metals/src/main/scala/scala/meta/internal/builds/ScalaCliBuildTool.scala

```diff
@@ -4,13 +4,51 @@ import java.security.MessageDigest
 
 import scala.concurrent.Future
 
+import scala.meta.internal.bsp.BspConfigGenerationStatus._
 import scala.meta.internal.metals.BuildInfo
 import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.UserConfiguration
 import scala.meta.internal.metals.scalacli.ScalaCli
 import scala.meta.io.AbsolutePath
 
-class ScalaCliBuildTool(val workspaceVersion: Option[String])
-    extends BuildTool {
+class ScalaCliBuildTool(
+    val workspaceVersion: Option[String],
+    userConfig: () => UserConfiguration,
+) extends BuildTool
+    with BuildServerProvider {
+
+  lazy val runScalaCliCommand: Option[Seq[String]] =
+    ScalaCli.localScalaCli(userConfig())
+
+  override def generateBspConfig(
+      workspace: AbsolutePath,
+      systemProcess: List[String] => Future[BspConfigGenerationStatus],
+  ): Future[BspConfigGenerationStatus] =
+    if (runScalaCliCommand.nonEmpty) {
+      systemProcess(createBspFileArgs(workspace))
+    } else {
+      // fallback to creating `.bsp/scala-cli.json` that starts JVM launcher
+      val bspConfig = workspace.resolve(".bsp").resolve("scala-cli.json")
+      bspConfig.writeText(ScalaCli.scalaCliBspJsonContent())
+      Future.successful(Generated)
+    }
+
+  def createBspConfigIfNone(
+      workspace: AbsolutePath,
+      systemProcess: List[String] => Future[BspConfigGenerationStatus],
+  ): Future[BspConfigGenerationStatus] = {
+    if (ScalaCliBuildTool.pathsToScalaCliBsp(workspace).exists(_.isFile))
+      Future.successful(Generated)
+    else generateBspConfig(workspace, systemProcess)
+  }
+
+  override def createBspFileArgs(workspace: AbsolutePath): List[String] =
+    runScalaCliCommand.getOrElse(Seq("scala-cli")).toList ++ List(
```

Comment:
We shouldn't do `.getOrElse(Seq("scala-cli"))`, if runScalaCliCommand is empty it means we didn't find scala-cli locally, so this will fail.

Reply:
I know. Should I return an empty list then?

Reply:
We would probably need something like an option of list here maybe? And handle it properly?